ADAL/src/utils/ADClientCapabilitiesUtil.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import <Foundation/Foundation.h>
+
+@interface ADClientCapabilitiesUtil : NSObject
+
++ (NSArray<NSString *> *)knownCapabilities:(NSArray<NSString *> *)allCapabilities;
+
+/*
+    Takes a list of capabilities and returns the JSON claims.
+    The result JSON is not URL encoded and caller needs to encode it if necessary
+ */
++ (NSString *)claimsRequestFromCapabilities:(NSArray<NSString *> *)capabilities;
+
+/*
+    Takes a list of capabilities and returns the JSON claims, combining them with any claims passed by developer.
+    The result JSON is not URL encoded and caller needs to encode it if necessary
+ */
+
++ (NSString *)claimsRequestFromCapabilities:(NSArray<NSString *> *)capabilities
+                            developerClaims:(NSDictionary *)developerClaims;
```

Comment:
Does `claimsRequestFromCapabilities:` sound like it is returning a request? :)
Will `requestClaimsFromCapabilities:` be better?

Reply:
What do you think about something like "claimsParameterFromCapabilities"?

Reply:
Sounds precise :)